### PR TITLE
raftstore/coprocessor: update log and key format

### PR DESCRIPTION
**Pull request by Connor1996**
Signed-off-by: Connor1996 <zbk602423539@gmail.com>

## What have you changed? (mandatory)
Update the output log format

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No need

## Does this PR affect documentation (docs) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

File: src/raftstore/coprocessor/split_check/keys.rs [hunk 1]
```diff
@@ -147,18 +148,19 @@ impl<C: Sender<Msg> + Send> SplitCheckObserver for KeysCheckObserver<C> {
...
         REGION_KEYS_HISTOGRAM.observe(region_keys as f64);
         if region_keys >= self.region_max_keys {
             info!(
-                "[region {}] approximate keys {} >= {}, need to do split check",
-                region.get_id(),
-                region_keys,
-                self.region_max_keys
+                "approximate keys over threshold, need to do split check";
+                "region_id" => region.get_id(),
+                "keys" => region_keys,
```
Comment (huachaohuang) [inline src/raftstore/coprocessor/split_check/keys.rs]: We should use hex here?

File: src/raftstore/coprocessor/split_observer.rs [hunk 2]
```diff
@@ -79,19 +79,24 @@ impl SplitObserver {
...
                             warn!(
-                                "[region {}] key {} is not larger than previous {} at {}, skip.",
-                                region_id,
-                                escape(&key),
-                                escape(last_valid_key.as_ref().unwrap()),
-                                k
+                                "key is not larger than previous, skip.";
+                                "region_id" => region_id,
+                                "key" => hex::encode_upper(key),
+                                "previous" => hex::encode_upper(last_valid_key.as_ref().unwrap()),
```
Comment (huachaohuang) [inline src/raftstore/coprocessor/split_observer.rs]: Please refer to this PR https://github.com/tikv/tikv/pull/4087.

File: src/raftstore/coprocessor/split_check/keys.rs (hunk 1 above)
Comment (Connor1996) [reply src/raftstore/coprocessor/split_check/keys.rs]: this is just a number

File: src/raftstore/coprocessor/split_observer.rs (hunk 2 above)
Comment (Connor1996) [reply src/raftstore/coprocessor/split_observer.rs]: the `Display` trait is implemented for `Key`, but here is `Vec<u8>`. Should I replace `Vec<u8>` with `Key`?

File: src/raftstore/coprocessor/region_info_accessor.rs [hunk 3]
```diff
@@ -226,7 +226,10 @@ impl RegionCollector {
         // receive an `Update` message, the region may have been deleted for some reason. So we
         // handle it according to whether the region exists in the collection.
         if self.regions.contains_key(&region.get_id()) {
-            info!("region_collector: trying to create region {} but it already exists, try to update it", region.get_id());
+            info!(
+                "region_collector: trying to create region but it already exists, try to update it";
```
Comment (overvenus) [inline src/raftstore/coprocessor/region_info_accessor.rs]:
```suggestion
                "trying to create this region but it already exists, try to update it";
```

Looks like we dont need to print `region_collector: ` as it is already printed by the file name.

Comment (AndreMouche) [reply src/raftstore/coprocessor/region_info_accessor.rs]: Is `region_collector` a key word for the monitor?

File: src/raftstore/coprocessor/split_observer.rs (hunk 2 above)
Comment (breeswish) [reply src/raftstore/coprocessor/split_observer.rs]: No you should use facilities in log_wrapper.

**Comment by huachaohuang** [comment]
@Connor1996 please drive this PR to get it merged.

**Comment by Connor1996** [comment]
PTAL

**Comment by huachaohuang** [comment]
@overvenus PTAL

File: src/raftstore/coprocessor/split_observer.rs [hunk 4]
```diff
@@ -79,19 +79,24 @@ impl SplitObserver {
...
                         if last_valid_key.as_ref().map_or(false, |k| *k >= key) {
                             warn!(
-                                "[region {}] key {} is not larger than previous {} at {}, skip.",
-                                region_id,
-                                escape(&key),
-                                escape(last_valid_key.as_ref().unwrap()),
-                                k
+                                "key is not larger than previous, skip.";
+                                "region_id" => region_id,
+                                "key" => ::log_wrappers::Key(&key),
```
Comment (breeswish) [inline src/raftstore/coprocessor/split_observer.rs]: we don't need the `::` prefix in Rust 2018 now.

**Comment by Connor1996** [comment]
PTAL again @huachaohuang @breeswish

**Pull request merged**